src/main/drivers/vtx_common.c

```diff
@@ -50,10 +80,54 @@ bool vtxCommonDeviceRegistered(void)
 void vtxCommonProcess(uint32_t currentTimeUs)
 {
     if (!vtxDevice)
-        return;
-
-    if (vtxDevice->vTable->process)
+    return;
+
+    static uint32_t lastCycleTimeUs;
+    static uint8_t scheduleIndex;
+    uint8_t currentSchedule = vtxParamSchedule[scheduleIndex];
+
+    if (vtxDevice->vTable->process) {
+        // Process VTX changes from the parameter group at 10Hz
+        if (currentTimeUs > lastCycleTimeUs + VTX_PARAM_CYCLE_TIME_US) {
+            const vtxSettingsConfig_t settings = vtxCommonGetSettings();
+            switch (currentSchedule)
```

Comment:
What is the benefit of splitting the processing of the changes here? Is it e.g. to avoid filling up the command queue of the SA implementation?

Reply:
Kind of a preventative measure to ensure the safety of the underlying buffers.